pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,311 @@
+            Vij = interaction_coeff / (abs(atom_distance) ** 6)  # van der Waals potential
+            coeffs.append(Vij)
+            observables.append(qml.prod(qml.Projector([1], wire1), qml.Projector([1], wire2)))
+            # Rydberg projectors
+
+    return RydbergHamiltonian(
+        coeffs, observables, register=register, interaction_coeff=interaction_coeff
+    )
+
+
+def rydberg_drive(amplitude, detuning, phase, wires):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the action of a driving laser
+    field with the given rabi frequency, detuning and phase acting on the given wires
+
+    .. math::
+
+        \frac{1}{2} \Omega(t) \sum_i (\cos(\phi)\sigma_i^x - \sin(\phi)\sigma_i^y) -
+        \frac{1}{2} \delta(t) \sum_i \sigma_i^z
+
+    where :math:`\Omega`, :math:`\delta` and :math:`\phi` correspond to the rabi frequency, detuning
+    and phase of the laser, :math:`i` correspond to the wire index, and :math:`\sigma^\alpha` for
+    :math:`\alpha = x,y,z` are the Pauli matrices. This driving term can be combined with an interaction
+    term to create a Hamiltonian describing a driven Rydberg atom system.
+
+    Args:
+        amplitude (Union[float, Callable]): float or callable returning the amplitude (in MHz) of a
+            laser field
+        detuning (Union[float, Callable]): float or callable returning the detuning (in MHz) of a
+            laser field
+        phase (float): float containing the phase (in radians) of the laser field
+        wires (Union[int, List[int]]): integer or list containing wire values for the Rydberg atoms that
+            the laser field acts on
+
+    Returns:
+        RydbergHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the action of the laser field
+        on the Rydberg atoms.
+
+    .. seealso::
+
+        :func:`~.rydberg_interaction`, :class:`~.ParametrizedHamiltonian`, :class:`~.ParametrizedEvolution`
+        and :func:`~.evolve`
+
+    **Example**
+
+    We create a Hamiltonian describing a laser acting on 4 wires (Rydberg atoms) with a fixed detuning and
+    phase, and a parametrized, time-dependent amplitude. The Hamiltonian includes an interaction term for
+    inter-atom interactions due to the Rydberg blockade, as well as the driving term for the laser driving
+    the atoms:
+
+    .. code-block:: python
+
+        amplitude = lambda p, t: p * jnp.sin(jnp.pi * t)
+        detuning = 3 * jnp.pi / 4
+        phase = jnp.pi / 2
+        wires = [0, 1, 2, 3]
+        H_d = qml.pulse.rydberg_drive(amplitude, detuning, phase, wires)
+
+        atom_coordinates = [[0, 0], [0, 4], [4, 0], [4, 4]]
+        H_i = qml.pulse.rydberg_interaction(atom_coordinates, wires)
+
+    >>> H_d
+    ParametrizedHamiltonian: terms=2
+    >>> H_i
+    ParametrizedHamiltonian: terms=6
+
+    The driving Hamiltonian has only two terms because it is simplified such that each term
+    represents one of the summations in the Hamiltonian shown above.
+
```

Comment:
```suggestion
    The two terms of the drive field correspond to the first and second sum, corresponding to the drive and the shift term.

```

Reply:
This behaviour is going to change with the callable-phase PR anyway